File: 图片分享/springboot（后端）/share/src/main/java/com/img/share/pojo/Comment.java
package com.img.share.pojo;


import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Comment {
    private Integer cid;
    private Img img;
    private String commdate;
    private User user;
    private int  clicklike;
    private String content;
    private List<User> userComments;
}

File: 图片分享/nestjs（管理系统后端)/img-share-manage/src/entity/comment.ts
import { BaseEntity, Column, Entity, JoinColumn, ManyToMany, ManyToOne, PrimaryGeneratedColumn } from "typeorm";
import { Img } from "./img";
import { User } from "./user";

@Entity()
export class Comment extends BaseEntity {
    @PrimaryGeneratedColumn()
    cid: number
    @ManyToOne(() => Img)
    @JoinColumn({
        name: 'iid',
        referencedColumnName: 'iid'
    })
    iid: Img
    @Column('datetime')
    commdate: string
    @ManyToOne(()=>User)
    @JoinColumn({
        name: 'uid',
        referencedColumnName: 'uid'
    })
    uid:User
    @Column()
    clicklike: number
    @Column('text')
    content: string
    
    @ManyToMany(() => User, users => users.likecomments)
    users:User[]
}